datadog/dogstatsd/container.py

```diff
@@ -55,3 +76,30 @@ def _read_container_id(self, fpath):
         except Exception as e:
             raise UnresolvableContainerID("Unable to read the container ID: " + str(e))
         return None
+
+    def _get_cgroup_from_inode(self):
+        """Read the container ID from the cgroup inode."""
+        # Parse /proc/self/cgroup and get a map of controller to its associated cgroup node path.
+        cgroup_controllers_paths = {}
+        with open(self.CGROUP_PATH, mode="r") as fp:
+            for line in fp:
+                tokens = line.strip().split(":")
+                if len(tokens) != 3:
+                    continue
+                if tokens[1] == self.CGROUPV1_BASE_CONTROLLER or tokens[1] == "":
+                    cgroup_controllers_paths[tokens[1]] = tokens[2]
+
+        # Retrieve the cgroup inode from "/sys/fs/cgroup + controller + cgroupNodePath"
+        for controller in [
+            self.CGROUPV1_BASE_CONTROLLER,
+            self.CGROUPV2_BASE_CONTROLLER,
+        ]:
+            if controller in cgroup_controllers_paths:
```

Comment:
```suggestion
            if controller, cpath in cgroup_controllers_paths.values():
```

Reply:
Actually wait, this is a `if`, not a `for`. This cannot work right?

Reply:
I've removed this.